### PR TITLE
[Github #672] improvements

### PR DESCRIPTION
**Pull request by khaliqgant**
# CLI
* Allow a dynamic port to be run when running `npx nango dev` via the `NANGO_PORT` env variable
* When running a command if the `nango-integrations` directory is missing prompt the user if they want to create default files so the command can proceed -- this would run `npx nango i && npx nango g`
* ~Add `configDeploy` which iterates over the `nango.yaml` and if any new syncs have been added to the provider immediately run those against those provider connections~
* ~[ ] if a sync has been removed, deleted it and remove the schedule~
* Add option to not compile interfaces on the `npx nango dev` command. This is useful if a dev is manually updating their `models.ts` file
* Add an .env value of `NANGO_AUTO_UPGRADE` which doesn't prompt the user to upgrade but just does so automatically
* Fix passing in a `lastSyncDate` value to the `npx nango sr` command

# Worker
* Store the `SyncJob.result` keyed by models so that an accurate count of added or updated records is set by a model. This fixes the webhook not firing if `nango.batchSend` is used and no returned records are at the end of the integration script
* Add a `'NANGO_INTEGRATIONS_FULL_PATH` env variable so that integrations don't have to be symlinked but rather can be set via an env variable

# Webapp
* Safer parsing if the activity messages are null for some reason
* Adjust parsing of the SyncJob.result to account for the changed JSON schema. Supports a keyed model or just an `{ added: number, updated: number } `

### CODE REVIEW

File: packages/cli/lib/sync.ts [hunk 1]
```diff
@@ -34,22 +47,179 @@ interface RunArgs {
...
+                syncName,
+                providerConfigKey,
+                runs,
+                returns
+            };
+
+            postData.push(body);
+        }
+    }
+    console.log(url);
```
Comment (bastienbeurier) [inline packages/cli/lib/sync.ts]: Intentional logs?

File: packages/cli/lib/sync.ts [hunk 2]
```diff
@@ -34,22 +47,179 @@ interface RunArgs {
...
+    console.log(postData);
+
+    await axios
+        .post(url, postData, { headers: enrichHeaders(), httpsAgent: httpsAgent() })
+        .then((response: any) => {
+            console.log(response.data);
+            //console.log(chalk.green(`Syncs has been updated and is now running`));
+        })
+        .catch((err) => {
+            console.log(err);
```
Comment (bastienbeurier) [inline packages/cli/lib/sync.ts]: Intentional unformatted log?

**Review by bastienbeurier — APPROVED** [review]

File: packages/cli/lib/sync.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/cli/lib/sync.ts]: Thanks, removed

File: packages/cli/lib/sync.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/cli/lib/sync.ts]: Updated, thanks!

**Pull request merged**